### PR TITLE
feat: expose contract to API

### PR DESCRIPTION
**Pull request by vildead**

### CODE REVIEW

**Comment by vildead** [comment]
@jLebioda could you please have a quick look? Still some things are missing and I will have to update serializer. But any comment at this stage will be very welcomed :)

File: web/views/api.py [hunk 1]
```diff
@@ -147,6 +149,38 @@ def datasets(request):
...
+def is_valid_field_in_model(klass_name, field_name):
+    getattr(klass_name, field_name, False)
+
+def get_filtered_entities(request, model_name):
+    filters = {}
+    for filter_key in request.GET:
+        if not is_valid_field_in_model(model_name, filter_key): continue
+        filters[filter_key] = request.GET.get(filter_key)
+
+    return getattr(sys.modules['core.models'], model_name).objects.filter(**filters)
```
Comment (jLebioda) [inline web/views/api.py:162]: this is super generic :) 
I actually thought more about passing a concrete `Contract` class to the function (see line 168), but this is even better!

File: core/models/contract.py [hunk 2]
```diff
@@ -136,7 +136,7 @@ def to_dict(self):
         base_dict = {
             'id': self.id,
             'comments': self.comments,
-            'project': self.project,
+            'project': self.project.id,
```
Comment (jLebioda) [inline core/models/contract.py:139]: are we sure that we won't break any exports by this change?

File: web/views/api.py [hunk 3]
```diff
@@ -15,9 +16,10 @@
 
 from stronghold.decorators import public
 
+from core.importer.contract_exporter import ContractExporter
 from core.importer.datasets_exporter import DatasetsExporter
 from core.importer.projects_exporter import ProjectsExporter
-from core.models import User, Cohort, Dataset, Partner, Project, DiseaseTerm
+from core.models import User, Cohort, Dataset, Partner, Project, DiseaseTerm, Contract
```
Comment (jLebioda) [inline web/views/api.py:22]: `Contract` seems not to be used anywhere

**Review by jLebioda — COMMENTED** [review]
Looks great! I didn't test it though yet

File: web/views/api.py (hunk 1 above)
Comment (vildead) [reply web/views/api.py:162]: I am not really sure we should be that generic :D just a bad feeling it makes things too loose.

File: core/models/contract.py (hunk 2 above)
Comment (vildead) [reply core/models/contract.py:139]: this should be ok - contract was not used in exports yet.

File: web/views/api.py (hunk 3 above)
Comment (vildead) [reply web/views/api.py]: fixed

File: web/views/projects.py [hunk 4]
```diff
@@ -281,3 +281,14 @@ def unpublish_project(request, pk):
     project.is_published = False
     project.save()
     return redirect(reverse_lazy('project', kwargs={'pk': project.id}))
+
+
+def dsw_list_projects(request):
+    #if data steward or admin -> list all the public projects
+    # if(request.user.is_admin() | request.user.is_datasteward()):
+    #     objects = Project.objects.all().filter(is_published=True)
+    objects = Project.objects.filter(local_custodians=request.user) | Project.objects.filter(company_personnel=request.user)
```
Comment (jLebioda) [inline web/views/projects.py:290]: from the efficiency point of view, it's better to add explicitly `is_published=True` to both queries here and remove line 291, or to move the line 291 before 290

**Comment by vildead** [comment]
consider creating independent package (installable via `pip`)

**Comment by vildead** [comment]
@jLebioda can you please review?
then we can merge and create new issue for moving the functionality to new package

**Review by jLebioda — APPROVED** [review]
looks good to me!

**Pull request merged**